utils/certs.go

```diff
@@ -93,7 +93,7 @@ func GenerateCert(hosts []string, certFile, keyFile, caFile, caKeyFile, org stri
 	}
 	// client
 	if len(hosts) == 1 && hosts[0] == "" {
-		template.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth}
+		template.ExtKeyUsage = []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth}
```

Comment:
Yes, I was actually seeing this in testing.  I will dig into the keys more.

As for the none driver, the way it's setup is to create the server keys per machine.  I don't think we want to re-use a separate set of server keys. 

So for the server certs to work with swarm, they just need the IP SAN of the swarm master right or am I misunderstanding?